language/polkavm-move-native/src/types.rs

```diff
@@ -51,22 +51,45 @@ pub struct MoveSignerVector {
 ///
 /// The pointer must be to static memory and never mutated.
 #[repr(C)]
+#[derive(Clone, Copy)]
```

Comment:
We probably gonna need two falvours of same move data type - one which is statically defined by LLVM in RO memory of guest, another one - Rust host type which is converted by reading the guest memory and doing all follows on pointers to other guest memory etc.

Reply:
Can you explain?

Reply:
When you build guest program with LLVM you define move types on the fly. For example MoveType has type_info pointer which points to another location of guest memory. So this type can be used inside guest program rust code, because pointer is in guest memory namespace so it can be directly followed. However on host mode - when you read MoveType from guest memory (load_from) - you will need to follow pointer to read additional data, but you cannot directly return MoveType, because pointer is invalid, you need to explicitly follow pointer and read that, put it into owned value and construct something like HostMoveType which has all values assembled as owned types, instead of raw pointers to guest memory.

Reply:
Hmm, yes. For now we only use the `type_desc` field on host side. I'll make a comment in the code.